helm/examples/efs/README.md

```diff
@@ -1,30 +1,113 @@
+```console
+kubectl config get-clusters | cut -d'.' -f1
+```
+Set EKS cluster name to variable "EKS_NAME"
+
+- EFS security groups
+```console
+  aws efs describe-mount-targets --file-system-id <EFS ID> --region <EFS region> |  awk -F $'\t' '{print $7}'
+  # For each file-system mount target:
+  aws efs describe-mount-targets-security-groups --mount-target-id <mount target id>  --region <EFS region> |  awk -F $'\t' '{print $2}'
+```
+Add each unique security group id to the variable "EFS_SECURITY_GROUPS"
 
-## Directory contents
+There is a setting "EFS_CSI_DRIVER", the default is "true" and it is recommended to leave that.
 
-### hpcc-efs/
-The hpcc-efs helm chart will provision an EFS Server Pod. It will also create a Persistent Volume Claim for each of the required HPCC storage types.
+## Deploy efs-provisioner
+  efs-provisioner pod should be started first:
+```console
+  ./install-efs-provision.sh
+  # To check the pod:
+  kubectl get pod
+```
+There may be some warnings which can be ignored:
+```code
+An error occurred (InvalidPermission.Duplicate) when calling the AuthorizeSecurityGroupIngress operation: the specified rule "peer: sg-0a5a005489115aac6, TCP, from port: 2049, to port: 2049, ALLOW" already exists
+```
 
-EFS Server should be created on AWS EFS service first.  The hpcc-efs helm chart needs an EFS region and id to configure the deployment. The default region is us-east-1.
+```code
+Warning: storage.k8s.io/v1beta1 CSIDriver is deprecated in v1.19+, unavailable in v1.22+; use storage.k8s.io/v1 CSIDriver
+csidriver.storage.k8s.io/efs.csi.aws.com configured
+```
+"v1beta" will be replaced with "v1" when it is available.
 
-Example use:
 
-helm install efsstorage examples/efs/hpcc-efs --set efs.region=<EFS region>  --set efs.id=<EFS ID>
-helm install myhpcc hpcc/ --set global.image.version=latest -f examples/efs/values-efs.yaml
+## Deploy HPCC cluster with values-auto-efs.yaml
+It will automically create Persistent Volume Claims (PVC) and delete them when the HPCC cluster is deleted:<br/>
+Under the helm directory:
+```console
+  helm install myhpcc ./hpcc --set global.image.version=<HPCC Platform version> -f examples/efs/values-auto-efs.yamls
+```
 
+## Deploy HPCC cluster with values-retained-efs.yaml
+This will require deploying HPCC PVCs first. PVCs will persist after the HPCC cluster is deleted.<br/>
+Under the helm directory:
+```console
+  helm install awsstorage examples/efs/hpcc-efs
+  # To start HPCC cluster:
+  helm install myhpcc hpcc/ --set global.image.version=latest -f examples/efs/values-retained-efs.yaml
+```
+helm uninstall will not delete EFS persistant volumes claims (PVC). You can either run "kubectl delete pv <pv name> or --all".
 
-### values-efs.yaml
-An example values file to be supplied when installing the HPCC chart.
-NB: Either use the output auto-generated when installing the "hpcc-efs" helm chart, or ensure the names in your values files for the storage types match the PVC names created.
```

Comment:
The README should still at least mention (as with other storage examples), the fact that the values-retained... is just an example of what would be generated as output from installing hpcc-efs.
i.e. if users want to use a different name in place of 'awsstorage' they can, the output is generated and will be correct, where as the example which is hard-wired to 'awsstorage' will not.

Reply:
Add following: 
An example values file to be supplied when installing the HPCC chart.
NB: Either use the output auto-generated when installing the "hpcc-efs" helm chart, or ensure the names in your values files for the storage types match the PVC names created. "values-retained-efs.yaml" expects that helm chart installation name is "awsstorage". Change the PVC name accordingly if another name is used.

Reply:
sounds fine.